src/components/HistoryTables/MsigProposals/Proposal.tsx

```diff
@@ -39,111 +35,96 @@ export default function ProposalDetail({
   cancel
 }: ProposalDetailProps) {
   const router = useRouter()
-  const { coinType, networkName } = useEnvironment()
   const logger = useLogger()
-  let {
+  const { coinType } = useEnvironment()
+
+  const [seeMore, setSeeMore] = useState(false)
+
+  // Ensure network cointype for address
+  const address = useMemo<string>(
+    () => convertAddrToPrefix(msigAddress, coinType),
+    [msigAddress, coinType]
+  )
+
+  // Get safe proposals
+  const {
     data: msigTxsData,
     loading: msigTxsLoading,
-    error: _msigTxsError
+    error: msigTxsError
   } = useMsigPendingQuery({
-    variables: {
-      address: convertAddrToPrefix(msigAddress, coinType)
-    },
+    variables: { address },
     pollInterval: 0
   })
 
-  const proposal = useMemo(() => {
-    if (!msigTxsLoading && !_msigTxsError) {
-      return msigTxsData?.msigPending.find(p => p.id === id)
-    }
-    return null
-  }, [msigTxsData, msigTxsLoading, _msigTxsError, id])
-
-  const [seeMore, setSeeMore] = useState(false)
+  // Find current proposal
+  const proposal = useMemo(
+    () => msigTxsData?.msigPending.find(p => p.id === id) || null,
+    [msigTxsData, id]
+  )
 
+  // Load the actor state
   const {
     data: actorData,
     loading: actorLoading,
+    notFound: actorNotFound,
     error: actorError
-  } = useActorQuery({
-    variables: { address: convertAddrToPrefix(msigAddress, coinType) }
-  })
```

Comment:
I'm not sure why to `useActorQuery` was used here. It seemed to only grab the actor code, to confirm that it was an msig actor and then perfom `useStateReadState`, which also gets the actor code